blockchain/blockindex.go

```diff
@@ -269,7 +269,8 @@ func newBlockIndex(db database.DB, chainParams *chaincfg.Params) *blockIndex {
 // This function is safe for concurrent access.
 func (bi *blockIndex) HaveBlock(hash *chainhash.Hash) bool {
 	bi.RLock()
-	_, hasBlock := bi.index[*hash]
+	node := bi.LookupNode(hash)
+	hasBlock := node != nil && node.status.HaveData()
```

Comment:
This line changes the behavior of the function `HaveBlock()` the documentation states that it returns whether a block exists in the `blockindex`. This line also asserts that the blockdata is also downloaded.

HaveBlock is only called in `blockchain.blockExists` in this repo anyways so why not edit that function instead? `HaveBlock` is exported so best not to change its behavior unless there's no alternative for backwards compatibility (for potential people using the library).

Reply:
Yeah this makes sense
I will make the required changes to blockExists and test the functionality